packages/cardpay-sdk/sdk/reward-manager/base.ts

```diff
@@ -308,6 +315,143 @@ export default class RewardManager {
     }
     return await waitForSubgraphIndexWithTxnReceipt(this.layer2Web3, txnHash);
   }
+
+  async addRewardRule(txnHash: string): Promise<TransactionReceipt>;
+  async addRewardRule(
+    prepaidCardAddress: string,
+    rewardProgramId: string,
+    blob: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt>;
+  async addRewardRule(
+    prepaidCardAddressOrTxnHash: string,
+    rewardProgramId?: string,
+    blob?: string,
+    txnOptions?: TransactionOptions,
+    contractOptions?: ContractOptions
+  ): Promise<TransactionReceipt> {
+    if (isTransactionHash(prepaidCardAddressOrTxnHash)) {
+      let txnHash = prepaidCardAddressOrTxnHash;
+      return await waitForSubgraphIndexWithTxnReceipt(this.layer2Web3, txnHash);
+    }
+    if (!prepaidCardAddressOrTxnHash) {
+      throw new Error('prepaidCardAddress is required');
+    }
+    if (!rewardProgramId) {
+      throw new Error('rewardProgramId is required');
+    }
+    if (!blob) {
+      throw new Error('blob is required');
+    }
+    let prepaidCardAddress = prepaidCardAddressOrTxnHash;
+    let { nonce, onNonce, onTxnHash } = txnOptions ?? {};
+    let from = contractOptions?.from ?? (await this.layer2Web3.eth.getAccounts())[0];
+
+    await this.checkPrepaidCardOwnerIsAdmin(rewardProgramId, prepaidCardAddress);
+
+    let gnosisResult = await executeSendWithRateLock(this.layer2Web3, prepaidCardAddress, async (rateLock) => {
+      let payload = await this.getAddRewardRulePayload(prepaidCardAddress, rewardProgramId, blob, rateLock);
+      if (nonce == null) {
+        nonce = getNextNonceFromEstimate(payload);
+        if (typeof onNonce === 'function') {
+          onNonce(nonce);
+        }
+      }
+      return await this.executeAddRewardRule(
+        prepaidCardAddress,
+        rewardProgramId,
+        blob,
+        rateLock,
+        payload,
+        await signPrepaidCardSendTx(this.layer2Web3, prepaidCardAddress, payload, nonce, from),
+        nonce
+      );
+    });
+
+    if (!gnosisResult) {
+      throw new Error(
+        `Unable to obtain a gnosis transaction result for lock reward program from prepaid card ${prepaidCardAddress}`
+      );
+    }
+
+    let txnHash = gnosisResult.ethereumTx.txHash;
+
+    if (typeof onTxnHash === 'function') {
+      await onTxnHash(txnHash);
+    }
+    return await waitForSubgraphIndexWithTxnReceipt(this.layer2Web3, txnHash);
+  }
+
+  async removeRewardProgram(txnHash: string): Promise<TransactionReceipt>;
+  async removeRewardProgram(
+    safeAddressIdOrTxnHash: string,
```

Comment:
when providing overloaded TS methods, the name of the params should reflect the specific overload--only the last implementation signature needs the "xxxORyyy" parameter. So this should be renamed to `safeAddress`